src/deploy.rs

```diff
@@ -409,23 +437,35 @@ pub async fn deploy_profile(
         let mut ssh_activate_child = ssh_activate_command
             .arg(self_activate_command)
             .spawn()
-            .map_err(DeployProfileError::SSHSpawnActivate)?;
+            .map_err(|err| {
+                DeployProfileError::SSHActivate(command::CommandError::OtherError(
+                    SSHActivateError::SpawnActivateError(err))
+                )
+            })?;
 
         if deploy_data.merged_settings.interactive_sudo.unwrap_or(false) {
             trace!("[activate] Piping in sudo password");
             handle_sudo_stdin(&mut ssh_activate_child, deploy_defs)
                 .await
-                .map_err(DeployProfileError::SSHActivatePipe)?;
+                .map_err(|err| {
+                    DeployProfileError::SSHActivate(command::CommandError::OtherError(
+                        SSHActivateError::ActivatePipeError(err)
+                    ))
+                })?;
         }
 
         let ssh_activate_exit_status = ssh_activate_child
-            .wait()
+            .wait_with_output()
```

Comment:
Is there a reason why the `run` method of `command::Command` is not used here instead of `wait_with_output`?

Reply:
Because `ssh_activate_child` is a `tokio::process::Child` and not a `command::Command` or `tokio::process::Command`. Although there are patterns, the way the codebase handles children is not repetitive enough to easily abstract them into a `run` command for a new `command::Child`, but I can put some more thought into this if you'd like.